src/components/PopoverWithMeasuredContent.js

```diff
@@ -59,138 +59,115 @@ const defaultProps = {
-        };
-
-        this.measurePopover = this.measurePopover.bind(this);
-    }
+function PopoverWithMeasuredContent(props) {
+    const {windowWidth, windowHeight} = useWindowDimensions();
+    const [popoverWidth, setPopoverWidth] = useState(props.popoverDimensions.width);
+    const [popoverHeight, setPopoverHeight] = useState(props.popoverDimensions.height);
+    const [isContentMeasured, SetIsContentMeasured] = useState(popoverWidth > 0 && popoverHeight > 0);
+    const [isVisible, SetIsVisible] = useState(false);
 
     /**
      * When Popover becomes visible, we need to recalculate the Dimensions.
      * Skip render on Popover until recalculations have done by setting isContentMeasured false as early as possible.
-     *
-     * @static
-     * @param {Object} props
-     * @param {Object} state
-     * @return {Object|null}
      */
-    static getDerivedStateFromProps(props, state) {
+    useEffect(() => {
         // When Popover is shown recalculate
-        if (!state.isVisible && props.isVisible) {
-            return {isContentMeasured: lodashGet(props, 'popoverDimensions.width', 0) > 0 && lodashGet(props, 'popoverDimensions.height', 0) > 0, isVisible: true};
+        if (!isVisible && props.isVisible) {
+            SetIsContentMeasured(lodashGet(props, 'popoverDimensions.width', 0) > 0 && lodashGet(props, 'popoverDimensions.height', 0) > 0);
+            SetIsVisible(true);
         }
         if (!props.isVisible) {
-            return {isVisible: false};
+            SetIsVisible(false);
         }
-        return null;
-    }
-
-    shouldComponentUpdate(nextProps, nextState) {
-        if (this.props.isVisible && (nextProps.windowWidth !== this.props.windowWidth || nextProps.windowHeight !== this.props.windowHeight)) {
-            return true;
-        }
-
-        // This component does not require re-render until any prop or state changes as we get the necessary info
-        // at first render. This component is attached to each message on the Chat list thus we prevent its re-renders
-        return !_.isEqual(_.omit(this.props, ['windowWidth', 'windowHeight']), _.omit(nextProps, ['windowWidth', 'windowHeight'])) || !_.isEqual(this.state, nextState);
-    }
+    }, [props, isVisible]);
 
     /**
      * Measure the size of the popover's content.
      *
      * @param {Object} nativeEvent
      */
-    measurePopover({nativeEvent}) {
-        this.popoverWidth = nativeEvent.layout.width;
-        this.popoverHeight = nativeEvent.layout.height;
-        this.setState({isContentMeasured: true});
-    }
+    const measurePopover = ({nativeEvent}) => {
+        setPopoverWidth(nativeEvent.layout.width);
+        setPopoverHeight(nativeEvent.layout.height);
+        SetIsContentMeasured(true);
+    };
 
     /**
      * Calculate the adjusted position of the popover.
      *
      * @returns {Object}
      */
-    calculateAdjustedAnchorPosition() {
+    const calculateAdjustedAnchorPosition = () => {
```

Comment:
The only usage is in line 138. Can we instead just wrap it in `useMemo` and store the value directly in `adjustedAnchorPosition`?